pvlib/location.py

```diff
@@ -276,3 +276,44 @@ def get_airmass(self, times=None, solar_position=None,
         airmass['airmass_absolute'] = airmass_absolute
 
         return airmass
+
+    def get_sun_rise_set_transit(self, times, method='pyephem', **kwargs):
+        """
+        Calculate sunrise, sunset and transit times.
+
+        Parameters
+        ----------
+        times : DatetimeIndex
+            Must be localized to the Location
+        method : str, default 'pyephem'
+            'pyephem', 'spa', or 'geometric'
+
+        kwargs are passed to the relevant functions. See
+        solarposition.sun_rise_set_transit_<method> for details.
+
+        Returns
+        -------
+        result : DataFrame
+            Column names are: ``sunrise, sunset, transit``.
+        """
+
+        if method == 'pyephem':
+            result = solarposition.sun_rise_set_transit_ephem(times,
+                                                              self.latitude,
+                                                              self.longitude,
+                                                              **kwargs)
+        elif method == 'spa':
+            result = solarposition.sun_rise_set_transit_spa(times,
+                                                            self.latitude,
+                                                            self.longitude,
+                                                            **kwargs)
+        elif method == 'geometric':
+            result = solarposition.sun_rise_set_transit_geometric(times,
+                                                            self.latitude,
+                                                            self.longitude,
+                                                            **kwargs)
+        else:
+            raise ValueError('{} is not a valid method. Must be '
+                             'one of pyephem, spa, geometric'
+                             .format(method))
+        return result
```

Comment:
I was thinking that here would be the place that you would convert the result to a dataframe, if a dataframe was passed in. IE: the low level functions return low level return values, while the high level function is more polite.

It makes sense to me to group functions in this hierarchy of low-level vs. high-level, because then the task of converting the result to a dataframe is DRY, it only occur once, rather than repeating WET code in multiple locations.

Reply:
That's a good catch. Changed to dataframe output, thanks!

Reply:
Hard to follow the history here. Is this comment still relevant or can it be marked as resolved? Github does not yet show it as outdated. It looks ok as of dd8d621

Reply:
Resolved.